### DISCUSSION TITLE
Error while Sending Transaction

### DISCUSSION THREAD

**Discussion by memxor** [aptos-labs/aptos-developer-discussions#150] [Questions]
### Describe your question in detail. For example how do you get all transactions for a user with the TypeScript SDK?

I'm getting the mentioned error while trying to Send a Transaction. My game was previously on Randomnet, but since it has been shut down, I moved the project to Devnet, and after moving it to devnet I'm getting this error. 

### What error, if any, are you getting?

{"message":"parse request payload error: failed to parse \"string(U64)\": invalid type: null, expected a string (occurred while parsing \"UserTransactionRequestInner\") (occurred while parsing \"EncodeSubmissionRequest\")","error_code":"web_framework_error","vm_error_code":null}

### What have you tried or looked at?  Or how can we reproduce the error?

I tried finding solutions online but couldn't find any. I've debugged the string I'm sending and have found no errors in that, and it is not null. 

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

Unity SDK

**Comment by 0xmaayan** [comment]
Hi @memxor could you share a code snippet?

**Reply by memxor** [reply]
Hi, yeah sure, here, I get a `ProtocolError 400`

```cs        
byte[] bytes = "3e79e6c4f4d55299f09b3aef9a8ba33a2ba0f53d081336c3811c3e4712a8d48b".ByteArrayFromHexString();
Sequence sequence = new(new ISerializable[] { new BString(username) });

EntryFunction payload = new(
    new(new AccountAddress(bytes), "aptos_horses_user"),
    "create_user",
    new(new ISerializableTag[] { }),
    sequence
);

Transaction createUserTx = new();
Coroutine createUser = StartCoroutine(RestClient.Instance.SubmitTransaction((_transaction, _responseInfo) =>
{
    createUserTx = _transaction;
    responseInfo = _responseInfo;
},
WalletManager.Instance.Wallet.Account,
payload));
yield return createUser;

if (responseInfo.status != ResponseInfo.Status.Success)
{
    Debug.LogError("Cannot create user. " + responseInfo.message);
    yield break;
}

Debug.Log("Transaction: " + createUserTx.Hash);
Coroutine waitForTransactionCor = StartCoroutine(
    RestClient.Instance.WaitForTransaction((_pending, _responseInfo) =>
    {
        responseInfo = _responseInfo;
    }, createUserTx.Hash)
);
yield return waitForTransactionCor;
```
